# PLAY STORE APPS ANALYSIS AND VISUALIZATION

## Task 1 : Pre-processing the data

Data Pre-processing is one of the important steps in data analytics because data that is not processed can lead to different unwanted results when the data will be used for 
further applications. This task includes sub-tasks such as handling null values, deletion or transformation of irrelevant values, datatype transformation, removing duplicates, 
etc. The tasks to be performed for cleaning the data set are given below:

In [ ]:
# Firstly, we would have to import all the necessary libraries to be utilized

import pandas as pd
import numpy as np
from numpy import nan
from datetime import datetime, timedelta


In [ ]:
#There are two datasets which we would then import using pandas
apps = pd.read_csv(r"C:\Users\anand\OneDrive\Desktop\Play_Store\Google_play_store.csv", index_col = 'App')
reviews = pd.read_csv(r"C:\Users\anand\OneDrive\Desktop\Play_Store\Google_play_store_user_reviews.csv", index_col = 'App')


In [ ]:
#Lets see the column attributes for apps dataset
apps.info()


<class 'pandas.core.frame.DataFrame'>
Index: 10841 entries, Photo Editor & Candy Camera & Grid & ScrapBook to iHoroscope - 2018 Daily Horoscope & Astrology
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Category        10841 non-null  object 
 1   Rating          9367 non-null   float64
 2   Reviews         10841 non-null  object 
 3   Size            10841 non-null  object 
 4   Installs        10841 non-null  object 
 5   Type            10840 non-null  object 
 6   Price           10841 non-null  object 
 7   Content Rating  10840 non-null  object 
 8   Genres          10841 non-null  object 
 9   Last Updated    10841 non-null  object 
 10  Current Ver     10833 non-null  object 
 11  Android Ver     10838 non-null  object 
dtypes: float64(1), object(11)
memory usage: 1.1+ MB


In [ ]:
# Let's find the no of rows and columns in apps dataset
apps.shape


(10841, 12)

In [ ]:
# Let's view the first 5 rows of the apps dataset
apps.head()


Category  Rating  \
App                                                                          
Photo Editor & Candy Camera & Grid & ScrapBook      ART_AND_DESIGN     4.1   
Coloring book moana                                 ART_AND_DESIGN     3.9   
U Launcher Lite – FREE Live Cool Themes, Hide Apps  ART_AND_DESIGN     4.7   
Sketch - Draw & Paint                               ART_AND_DESIGN     4.5   
Pixel Draw - Number Art Coloring Book               ART_AND_DESIGN     4.3   

                                                   Reviews  Size     Installs  \
App                                                                             
Photo Editor & Candy Camera & Grid & ScrapBook         159   19M      10,000+   
Coloring book moana                                    967   14M     500,000+   
U Launcher Lite – FREE Live Cool Themes, Hide Apps   87510  8.7M   5,000,000+   
Sketch - Draw & Paint                               215644   25M  50,000,000+   
Pixel Draw - Number Art Coloring Book                  967  2.8M     100,000+   

                                                    Type Price Content Rating  \
App                                                                             
Photo Editor & Candy Camera & Grid & ScrapBook      Free     0       Everyone   
Coloring book moana                                 Free     0       Everyone   
U Launcher Lite – FREE Live Cool Themes, Hide Apps  Free     0       Everyone   
Sketch - Draw & Paint                               Free     0           Teen   
Pixel Draw - Number Art Coloring Book               Free     0       Everyone   

                                                                       Genres  \
App                                                                             
Photo Editor & Candy Camera & Grid & ScrapBook                   Art & Design   
Coloring book moana                                 Art & Design;Pretend Play   
U Launcher Lite – FREE Live Cool Themes, Hide Apps               Art & Design   
Sketch - Draw & Paint                                            Art & Design   
Pixel Draw - Number Art Coloring Book                 Art & Design;Creativity   

                                                        Last Updated  \
App                                                                    
Photo Editor & Candy Camera & Grid & ScrapBook       January 7, 2018   
Coloring book moana                                 January 15, 2018   
U Launcher Lite – FREE Live Cool Themes, Hide Apps    August 1, 2018   
Sketch - Draw & Paint                                   June 8, 2018   
Pixel Draw - Number Art Coloring Book                  June 20, 2018   

                                                           Current Ver  \
App                                                                      
Photo Editor & Candy Camera & Grid & ScrapBook                   1.0.0   
Coloring book moana                                              2.0.0   
U Launcher Lite – FREE Live Cool Themes, Hide Apps               1.2.4   
Sketch - Draw & Paint                               Varies with device   
Pixel Draw - Number Art Coloring Book                              1.1   

                                                     Android Ver  
App                                                               
Photo Editor & Candy Camera & Grid & ScrapBook      4.0.3 and up  
Coloring book moana                                 4.0.3 and up  
U Launcher Lite – FREE Live Cool Themes, Hide Apps  4.0.3 and up  
Sketch - Draw & Paint                                 4.2 and up  
Pixel Draw - Number Art Coloring Book                 4.4 and up

In [ ]:
apps.duplicated().value_counts()


False    10349
True       492
Name: count, dtype: int64

### Findings on `apps` dataset

1. Data contains 10841 rows and 12 columns 
2. When importing the datasets the `App` column was used as the index column.
3. The `apps` dataset contains 492 duplicates rows and 10349 unique rows.
4. `Last Updated` column has a object datatype.
5. `Rating`,`Current Ver`, `Andriod Ver` `Type` columns contains missing values
6. Some columns contains irrelevant values that must be cleaned or removed.

### Subtask 1 : Removing Duplicate Rows

We have established that 492 columns in the apps dataset contains duplicate rows. We would then attempt to remove those duplicate rows from our apps dataset.

In [ ]:
# Drop all duplicate rows that appears more than once in 'apps' dataset while retaining its first row 
apps.drop_duplicates(keep='first', inplace=True)


In [ ]:
#Check to make sure duplicates rows were dropped
apps.info()


<class 'pandas.core.frame.DataFrame'>
Index: 10349 entries, Photo Editor & Candy Camera & Grid & ScrapBook to iHoroscope - 2018 Daily Horoscope & Astrology
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Category        10349 non-null  object 
 1   Rating          8891 non-null   float64
 2   Reviews         10349 non-null  object 
 3   Size            10349 non-null  object 
 4   Installs        10349 non-null  object 
 5   Type            10348 non-null  object 
 6   Price           10349 non-null  object 
 7   Content Rating  10348 non-null  object 
 8   Genres          10349 non-null  object 
 9   Last Updated    10349 non-null  object 
 10  Current Ver     10341 non-null  object 
 11  Android Ver     10346 non-null  object 
dtypes: float64(1), object(11)
memory usage: 1.0+ MB


### SubTask 2: Remove Irrelevant values from each column if any

We would check each column to ascertain attributes that are irrelevant and must be removed from the dataset.

We would start by checking the unique attributes for each column so as to source for irregularities.

In [ ]:
apps['Category'].unique()


array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

The `category` column loos okay except the 1.9 attribute as seen above which is not a google playstore category and thus we would attempt to remove the row entry containing the category of '1.9'

In [ ]:
#Check the row that contains the category of 1.9
apps[apps['Category'] == '1.9']


Category  Rating Reviews    Size  \
App                                                                        
Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M  1,000+   

                                        Installs Type     Price  \
App                                                               
Life Made WI-Fi Touchscreen Photo Frame     Free    0  Everyone   

                                        Content Rating             Genres  \
App                                                                         
Life Made WI-Fi Touchscreen Photo Frame            NaN  February 11, 2018   

                                        Last Updated Current Ver Android Ver  
App                                                                           
Life Made WI-Fi Touchscreen Photo Frame       1.0.19  4.0 and up         NaN

In [ ]:
# only one row (row 10472) contain the category of 1.9 and that row should be removed form our dataset since its not needed.

apps.drop('Life Made WI-Fi Touchscreen Photo Frame', inplace=True)


In [ ]:
# Check to make sure the category of 1.9 was removed.

apps['Category'].unique()


array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION'],
      dtype=object)

In [ ]:
# Check the distinct attributes of `Rating` column
apps['Rating'].unique()


array([4.1, 3.9, 4.7, 4.5, 4.3, 4.4, 3.8, 4.2, 4.6, 3.2, 4. , nan, 4.8,
       4.9, 3.6, 3.7, 3.3, 3.4, 3.5, 3.1, 5. , 2.6, 3. , 1.9, 2.5, 2.8,
       2.7, 1. , 2.9, 2.3, 2.2, 1.7, 2. , 1.8, 2.4, 1.6, 2.1, 1.4, 1.5,
       1.2])

The Rating column contains missing values. Since the Ratings for each app is a numerical value, these missing values will be replaced with 0.   

In [ ]:
#replace all nan values with 0
apps['Rating'] = apps['Rating'].fillna(0)


In [ ]:
# Check the Rating column to confirm that missing nan rows have been filled with 0
apps['Rating'].unique()


array([4.1, 3.9, 4.7, 4.5, 4.3, 4.4, 3.8, 4.2, 4.6, 3.2, 4. , 0. , 4.8,
       4.9, 3.6, 3.7, 3.3, 3.4, 3.5, 3.1, 5. , 2.6, 3. , 1.9, 2.5, 2.8,
       2.7, 1. , 2.9, 2.3, 2.2, 1.7, 2. , 1.8, 2.4, 1.6, 2.1, 1.4, 1.5,
       1.2])

In [ ]:
# Check the distinct attributes of `Reviews` column
apps['Reviews'].unique()


array(['159', '967', '87510', ..., '603', '1195', '398307'], dtype=object)

In [ ]:
# Check the distinct attributes of `Size` column
apps['Size'].unique()


array(['19M', '14M', '8.7M', '25M', '2.8M', '5.6M', '29M', '33M', '3.1M',
       '28M', '12M', '20M', '21M', '37M', '2.7M', '5.5M', '17M', '39M',
       '31M', '4.2M', '7.0M', '23M', '6.0M', '6.1M', '4.6M', '9.2M',
       '5.2M', '11M', '24M', 'Varies with device', '9.4M', '15M', '10M',
       '1.2M', '26M', '8.0M', '7.9M', '56M', '57M', '35M', '54M', '201k',
       '3.6M', '5.7M', '8.6M', '2.4M', '27M', '2.5M', '16M', '3.4M',
       '8.9M', '3.9M', '2.9M', '38M', '32M', '5.4M', '18M', '1.1M',
       '2.2M', '4.5M', '9.8M', '52M', '9.0M', '6.7M', '30M', '2.6M',
       '7.1M', '3.7M', '22M', '7.4M', '6.4M', '3.2M', '8.2M', '9.9M',
       '4.9M', '9.5M', '5.0M', '5.9M', '13M', '73M', '6.8M', '3.5M',
       '4.0M', '2.3M', '7.2M', '2.1M', '42M', '7.3M', '9.1M', '55M',
       '23k', '6.5M', '1.5M', '7.5M', '51M', '41M', '48M', '8.5M', '46M',
       '8.3M', '4.3M', '4.7M', '3.3M', '40M', '7.8M', '8.8M', '6.6M',
       '5.1M', '61M', '66M', '79k', '8.4M', '118k', '44M', '695k', '1.6M',
     

In [ ]:
# Check the distinct attributes of `Installs` column
apps['Installs'].unique()


array(['10,000+', '500,000+', '5,000,000+', '50,000,000+', '100,000+',
       '50,000+', '1,000,000+', '10,000,000+', '5,000+', '100,000,000+',
       '1,000,000,000+', '1,000+', '500,000,000+', '50+', '100+', '500+',
       '10+', '1+', '5+', '0+', '0'], dtype=object)

In [ ]:
# Check the distinct attributes of `Price` column
apps['Price'].unique()


array(['0', '$4.99', '$3.99', '$6.99', '$1.49', '$2.99', '$7.99', '$5.99',
       '$3.49', '$1.99', '$9.99', '$7.49', '$0.99', '$9.00', '$5.49',
       '$10.00', '$24.99', '$11.99', '$79.99', '$16.99', '$14.99',
       '$1.00', '$29.99', '$12.99', '$2.49', '$10.99', '$1.50', '$19.99',
       '$15.99', '$33.99', '$74.99', '$39.99', '$3.95', '$4.49', '$1.70',
       '$8.99', '$2.00', '$3.88', '$25.99', '$399.99', '$17.99',
       '$400.00', '$3.02', '$1.76', '$4.84', '$4.77', '$1.61', '$2.50',
       '$1.59', '$6.49', '$1.29', '$5.00', '$13.99', '$299.99', '$379.99',
       '$37.99', '$18.99', '$389.99', '$19.90', '$8.49', '$1.75',
       '$14.00', '$4.85', '$46.99', '$109.99', '$154.99', '$3.08',
       '$2.59', '$4.80', '$1.96', '$19.40', '$3.90', '$4.59', '$15.46',
       '$3.04', '$4.29', '$2.60', '$3.28', '$4.60', '$28.99', '$2.95',
       '$2.90', '$1.97', '$200.00', '$89.99', '$2.56', '$30.99', '$3.61',
       '$394.99', '$1.26', '$1.20', '$1.04'], dtype=object)

In [ ]:
# Check the distinct attributes of `Genres` column
apps['Genres'].unique()


array(['Art & Design', 'Art & Design;Pretend Play',
       'Art & Design;Creativity', 'Art & Design;Action & Adventure',
       'Auto & Vehicles', 'Beauty', 'Books & Reference', 'Business',
       'Comics', 'Comics;Creativity', 'Communication', 'Dating',
       'Education;Education', 'Education', 'Education;Creativity',
       'Education;Music & Video', 'Education;Action & Adventure',
       'Education;Pretend Play', 'Education;Brain Games', 'Entertainment',
       'Entertainment;Music & Video', 'Entertainment;Brain Games',
       'Entertainment;Creativity', 'Events', 'Finance', 'Food & Drink',
       'Health & Fitness', 'House & Home', 'Libraries & Demo',
       'Lifestyle', 'Lifestyle;Pretend Play',
       'Adventure;Action & Adventure', 'Arcade', 'Casual', 'Card',
       'Casual;Pretend Play', 'Action', 'Strategy', 'Puzzle', 'Sports',
       'Music', 'Word', 'Racing', 'Casual;Creativity',
       'Casual;Action & Adventure', 'Simulation', 'Adventure', 'Board',
       'Trivia', 'Role 

In [ ]:
# Check the `Current Ver` column for null values
apps[apps['Current Ver'].isna()]


Category  Rating Reviews  Size  \
App                                                                          
Learn To Draw Kawaii Characters       ART_AND_DESIGN     3.2      55  2.7M   
Market Update Helper              LIBRARIES_AND_DEMO     4.1   20145   11k   
Virtual DJ Sound Mixer                         TOOLS     4.2    4010  8.7M   
BT Master                                     FAMILY     0.0       0  222k   
Dots puzzle                                   FAMILY     4.0     179   14M   
Calculate My IQ                               FAMILY     0.0      44  7.2M   
UFO-CQ                                         TOOLS     0.0       1  237k   
La Fe de Jesus                   BOOKS_AND_REFERENCE     0.0       8  658k   

                                   Installs  Type  Price Content Rating  \
App                                                                       
Learn To Draw Kawaii Characters      5,000+  Free      0       Everyone   
Market Update Helper             1,000,000+  Free      0       Everyone   
Virtual DJ Sound Mixer             500,000+  Free      0       Everyone   
BT Master                              100+  Free      0       Everyone   
Dots puzzle                         50,000+  Paid  $0.99       Everyone   
Calculate My IQ                     10,000+  Free      0       Everyone   
UFO-CQ                                  10+  Paid  $0.99       Everyone   
La Fe de Jesus                       1,000+  Free      0       Everyone   

                                            Genres       Last Updated  \
App                                                                     
Learn To Draw Kawaii Characters       Art & Design       June 6, 2018   
Market Update Helper              Libraries & Demo  February 12, 2013   
Virtual DJ Sound Mixer                       Tools       May 10, 2017   
BT Master                                Education   November 6, 2016   
Dots puzzle                                 Puzzle     April 18, 2018   
Calculate My IQ                      Entertainment      April 3, 2017   
UFO-CQ                                       Tools       July 4, 2016   
La Fe de Jesus                   Books & Reference   January 31, 2017   

                                Current Ver Android Ver  
App                                                      
Learn To Draw Kawaii Characters         NaN  4.2 and up  
Market Update Helper                    NaN  1.5 and up  
Virtual DJ Sound Mixer                  NaN  4.0 and up  
BT Master                               NaN  1.6 and up  
Dots puzzle                             NaN  4.0 and up  
Calculate My IQ                         NaN  2.3 and up  
UFO-CQ                                  NaN  2.0 and up  
La Fe de Jesus                          NaN  3.0 and up

In [ ]:
# Check the count of null values in `Current Ver` column
apps['Current Ver'].isna().value_counts()


Current Ver
False    10340
True         8
Name: count, dtype: int64

In [ ]:
# Check the count of null values in `Andriod Ver` column
apps['Android Ver'].isna().value_counts()


Android Ver
False    10346
True         2
Name: count, dtype: int64

In [ ]:
# Check the distinct attributes of `Andriod Ver` column
apps['Android Ver'].unique()


array(['4.0.3 and up', '4.2 and up', '4.4 and up', '2.3 and up',
       '3.0 and up', '4.1 and up', '4.0 and up', '2.3.3 and up',
       'Varies with device', '2.2 and up', '5.0 and up', '6.0 and up',
       '1.6 and up', '1.5 and up', '2.1 and up', '7.0 and up',
       '5.1 and up', '4.3 and up', '4.0.3 - 7.1.1', '2.0 and up',
       '3.2 and up', '4.4W and up', '7.1 and up', '7.0 - 7.1.1',
       '8.0 and up', '5.0 - 8.0', '3.1 and up', '2.0.1 and up',
       '4.1 - 7.1.1', '5.0 - 6.0', '1.0 and up', '2.2 - 7.1.1',
       '5.0 - 7.1.1'], dtype=object)

In [ ]:
# Drop null values in `Andriod Ver` column
apps.dropna(subset=['Android Ver'], inplace=True)


In [ ]:
# Check the distinct attributes of `Content Rating` column
apps['Content Rating'].unique()


array(['Everyone', 'Teen', 'Everyone 10+', 'Mature 17+',
       'Adults only 18+', 'Unrated'], dtype=object)

In [ ]:
# Check for unique column attributes
apps['Type'].unique()


array(['Free', 'Paid', nan], dtype=object)

In [ ]:
# Drop rows with missing values in `Type` column
apps.dropna(subset=['Type'], inplace=True)


#### Check

Let's confirm that the nan row has been removed from the `Type` column

In [ ]:
apps['Type'].unique()


array(['Free', 'Paid'], dtype=object)

#### Next is to check for incorrect datatype in column attributes. The `Last Updated` column has a datatype of object and must be changed to a datatype of datetime to reflect the date attributes in it

In [ ]:
# Change datatype of `Last Updated` column to datetime

apps['Last Updated'] = pd.to_datetime(apps['Last Updated'])


In [ ]:
# Let's check that all changes made are reflected in every column.
apps.info()


<class 'pandas.core.frame.DataFrame'>
Index: 10345 entries, Photo Editor & Candy Camera & Grid & ScrapBook to iHoroscope - 2018 Daily Horoscope & Astrology
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Category        10345 non-null  object        
 1   Rating          10345 non-null  float64       
 2   Reviews         10345 non-null  object        
 3   Size            10345 non-null  object        
 4   Installs        10345 non-null  object        
 5   Type            10345 non-null  object        
 6   Price           10345 non-null  object        
 7   Content Rating  10345 non-null  object        
 8   Genres          10345 non-null  object        
 9   Last Updated    10345 non-null  datetime64[ns]
 10  Current Ver     10337 non-null  object        
 11  Android Ver     10345 non-null  object        
dtypes: datetime64[ns](1), float64(1), object(10)
memory usage: 1.0+ MB


### Summary of Data Cleaning on `Apps` Dataset

1. Removed all duplicate rows in the apps dataset.
2. Removed 1no. row containing '1.9' in the `Category` variable.
3. Replaced all missing values in `Rating` variable with 0.
4. Removed all rows containing missing values in `Andriod Ver` variable.
5. Removed all rows containing missing values in `Type` variable.
6. Changed the datatype of `Last Updated` variable to datetime.

From the above assessment, it can be seen that the `app` dataset still contains missing values in the `Current Ver` columns which will be addressed in the next stage of our analysis using Microsoft Excel. All missing values in the `Current Ver` column will be replaced with NaN.

#### Now we can export our `apps` dataset to csv file for further assessment/analysis

In [ ]:
apps.to_csv('cleaned_apps_v2.csv', encoding = 'utf=8')


### Data Preparation

The following data preparation processes was carried out on the cleaned apps dataset after exporting the cleaned apps dataset to csv.

1. Replaced all missing values in the `Current Ver` variable with NaN (Carried out on Microsoft Excel)
2. Cleaned the apps columns for special characters and irrelevant app names.

 After data preparation, the cleaned apps dataset was found to contain 9766 rows and 13 variables
 
 This cleaned apps dataset can now be analyzed to gain insight using MySQL and visualized using Power BI.

## Pre-processing on the `review` dataset

Lets programmatically view the reviews dataset to get familiar with its properties.

In [ ]:
# Check for dataet dimensions
reviews.shape


(64295, 4)

In [ ]:
# Lets see the column attributes for reviews dataset
reviews.info()


<class 'pandas.core.frame.DataFrame'>
Index: 64295 entries, 10 Best Foods for You to Houzz Interior Design Ideas
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Translated_Review       37427 non-null  object 
 1   Sentiment               37432 non-null  object 
 2   Sentiment_Polarity      37432 non-null  float64
 3   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(2)
memory usage: 2.5+ MB


In [ ]:
# Check dataset to view the first 10 rows
reviews.head(10)


Translated_Review  \
App                                                                        
10 Best Foods for You  I like eat delicious food. That's I'm cooking ...   
10 Best Foods for You    This help eating healthy exercise regular basis   
10 Best Foods for You                                                NaN   
10 Best Foods for You         Works great especially going grocery store   
10 Best Foods for You                                       Best idea us   
10 Best Foods for You                                           Best way   
10 Best Foods for You                                            Amazing   
10 Best Foods for You                                                NaN   
10 Best Foods for You                               Looking forward app,   
10 Best Foods for You              It helpful site ! It help foods get !   

                      Sentiment  Sentiment_Polarity  Sentiment_Subjectivity  
App                                                                          
10 Best Foods for You  Positive                1.00                0.533333  
10 Best Foods for You  Positive                0.25                0.288462  
10 Best Foods for You       NaN                 NaN                     NaN  
10 Best Foods for You  Positive                0.40                0.875000  
10 Best Foods for You  Positive                1.00                0.300000  
10 Best Foods for You  Positive                1.00                0.300000  
10 Best Foods for You  Positive                0.60                0.900000  
10 Best Foods for You       NaN                 NaN                     NaN  
10 Best Foods for You   Neutral                0.00                0.000000  
10 Best Foods for You   Neutral                0.00                0.000000

In [ ]:
# Check for missing values in the `Translated_Review` column
reviews['Translated_Review'].isna().value_counts()


Translated_Review
False    37427
True     26868
Name: count, dtype: int64

### Summary of finding on `reviews` dataset

1. `reviews` dataset contains 64295 rows and 4 variables
2. The dataset contains a lot of missing values for different reviews records
3. The `App` column was used as the index column

#### Sub-task - Drop all rows with missing values

In [ ]:
# Drop all missing values in the reviews dataaset
reviews.dropna(inplace=True)


We could observed that the `reviews` dataset had 26868 null values in its `Translated_Review` column. Therefore all rows with null values in the `Translated_Review` column was deleted as there is no translated reviews present and as such that kind of data is of no use for our analysis

#### Check

Check to make sure all missing values in the reviews dataset was dropped.

In [ ]:
# Check for any remaining missing values in `Transalated_Reviews` column
reviews['Translated_Review'].isna().value_counts()


Translated_Review
False    37427
Name: count, dtype: int64

In [ ]:
# Check for duplicates 
reviews.duplicated().value_counts()


False    27994
True      9433
Name: count, dtype: int64

In [ ]:
# Check dataset to confirm that all data cleaning has reflected.
reviews.info()


<class 'pandas.core.frame.DataFrame'>
Index: 37427 entries, 10 Best Foods for You to Housing-Real Estate & Property
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Translated_Review       37427 non-null  object 
 1   Sentiment               37427 non-null  object 
 2   Sentiment_Polarity      37427 non-null  float64
 3   Sentiment_Subjectivity  37427 non-null  float64
dtypes: float64(2), object(2)
memory usage: 1.4+ MB


_Duplicate records in the reviews dataset would not be dropped as all reviews records will be useful for further analysis_

### Summary of Data Cleaning on `reviews` dataset

1. All rows containing missing values were dropped from the dataset.
2. After dropping missing values the dataset now contains 37427 rows of data

The reviews dataset will now be exported as a csv file for further analysis

In [ ]:
reviews.to_csv('cleaned_reviews_v2.csv', encoding = 'utf=8')
